src/main/java/green/rijks/RijksFrame.java

```diff
@@ -0,0 +1,143 @@
+        setTitle("Rijks Museum Paintings");
+        setDefaultCloseOperation(EXIT_ON_CLOSE);
+
+        JPanel main = new JPanel();
+        main.setLayout(new BorderLayout());
+        setContentPane(main);
+
+        JButton next = new JButton("Next");
+        JButton prev = new JButton("Previous");
+
+        JPanel prevPanel = new JPanel(new FlowLayout(FlowLayout.LEFT));
+        JPanel nextPanel = new JPanel(new FlowLayout(FlowLayout.RIGHT));
+        prevPanel.add(prev);
+        nextPanel.add(next);
+
+        main.add(prevPanel, BorderLayout.WEST);
+        main.add(nextPanel, BorderLayout.EAST);
+
+        JTextField search = new JTextField();
+        main.add(search, BorderLayout.PAGE_START);
+
+        main.add(paintingComponent, BorderLayout.CENTER);
+
+        onStart();
+
+        next.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                Disposable disposable = service.getPage(key.get(), ++currPage)
+                        .subscribeOn(Schedulers.io())
+                        .observeOn(SwingSchedulers.edt())
+                        .subscribe(
+                                (response) -> handleResponse(response),
+                                Throwable::printStackTrace);
+            }
+        });
+
+        prev.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                Disposable disposable = service.getPage(key.get(), --currPage)
+                        .subscribeOn(Schedulers.io())
+                        .observeOn(SwingSchedulers.edt())
+                        .subscribe(
+                                (response) -> handleResponse(response),
+                                Throwable::printStackTrace);
+            }
+        });
+
+
+        DocumentListener docListener = new DocumentListener() {
+            @Override
+            public void insertUpdate(DocumentEvent e) {
+                Disposable disposable = service.getField(key.get(), search.getText(), --currPage)
+                        .subscribeOn(Schedulers.io())
+                        .observeOn(SwingSchedulers.edt())
+                        .subscribe(
+                                (response) -> handleResponse(response),
+                                Throwable::printStackTrace);
+            }
+
+            @Override
+            public void removeUpdate(DocumentEvent e) {
+
+            }
+
+            @Override
+            public void changedUpdate(DocumentEvent e) {
+
+            }
+        };
+
+    }
+
+    private void onStart() {
+        Disposable disposable = service.getPage(key.get(), currPage)
+                .subscribeOn(Schedulers.io())
+                .observeOn(SwingSchedulers.edt())
+                .subscribe(
+                        (response) -> handleResponse(response),
+                        Throwable::printStackTrace);
```

Comment:
maybe remove some of the duplicate code between this method and the ActionListeners for next and previous